Explorer/Assets/DCL/AvatarRendering/AvatarShape/Systems/AvatarShapeVisibilitySystem.cs

```diff
@@ -126,9 +131,24 @@ private void UpdateNonPlayerAvatarVisibilityOnCameraDistance(in AvatarCustomSkin
         }
 
         [Query]
-        private void UpdateAvatarsVisibilityState(ref AvatarShapeComponent avatarShape, ref AvatarCachedVisibilityComponent avatarCachedVisibility)
+        private void BlockAvatars(in Entity entity, ref AvatarShapeComponent avatarShapeComponent)
         {
-            bool shouldBeHidden = avatarShape.HiddenByModifierArea;
+            if (!userBlockingCacheProxy.Configured) return;
+
+            if (avatarShapeComponent.InstantiatedWearables.Count == 0) return;
+
+            bool isBlocked = userBlockingCacheProxy.Object!.UserIsBlocked(avatarShapeComponent.ID);
+
+            if (isBlocked && !World.Has<BlockedPlayerComponent>(entity))
+                World.Add(entity, new BlockedPlayerComponent());
+            else if (!isBlocked && World.Has<BlockedPlayerComponent>(entity))
+                World.Remove<BlockedPlayerComponent>(entity);
```

Comment:
Very minor thing, I don't know how much of an improvement could be but I think that you could simplify the code here for better readability by using the World.TryRemove and removing the Has statement in the if